src/components/SubpageUpload/PageUpload.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+import { useState } from 'react';
+import UploadService from "./upload-files.service";
+import Backdrop from '@material-ui/core/Backdrop';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Snackbar from '@material-ui/core/Snackbar';
+import Alert from '@material-ui/lab/Alert';
+//import * as musicMetadata from 'music-metadata-browser';
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+
+    },
+    input: {
+        display: 'none',
+    },
+    backdrop: {
+        zIndex: theme.zIndex.drawer + 1,
+        color: '#fff',
+    },
+}));
+
+export default function PageUpload() {
+    const classes = useStyles();
+    const [uploadMessage, setUploadMessage] = useState(0);
+    const [uploadProgress, setUploadProgress] = useState(-1);
+    const [backdrop, setBackdrop] = useState(false); //false means enabled
+    const [open, setOpen] = React.useState(false);
+
+    const handleClose = (event, reason) => {
+        if (reason === 'clickaway') {
+            return;
+        }
+        setOpen(false);
+    };
+
+    let upload = (e) => {
```

Comment:
use `const` instead of `let`

The whole method has to be refactored. It is completely unreadable with this cyclic complexity.
I'd strongly recommend to use [promise chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) to do all the metadata checks and reject the promise with the error message so that the `.catch(e)` can then set the correct error message. The last element in the chain will be the upload itself. There is no need to nest promises. The error of the POST can then also be handled in the same catch.

Reply:
You might have to show/explain it to me in person. I do not understand the conecept yet.